### PR TITLE
make create_join_table compatible.

### PR DESCRIPTION
**Pull request by kuzukuzu**
Follow up of https://github.com/rails/rails/pull/28217 .
Fix https://github.com/rails/rails/issues/30654 .

### CODE REVIEW

**Comment by kamipo** [comment]
Can you change this PR target against master branch?

**Comment by kuzukuzu** [comment]
Do it. thx.

File: activerecord/lib/active_record/migration/compatibility.rb [hunk 1]
```diff
@@ -71,6 +71,19 @@ class << t
...
+        def create_join_table(table_1, table_2, column_options: {}, **options)
+          if block_given?
+            super(table_1, table_2, column_options: column_options.reverse_merge!(type: :integer), **options) do |t|
+              class << t
+                prepend TableDefinition
+              end
+              yield t
+            end
+          else
+            super(table_1, table_2, column_options: column_options.reverse_merge!(type: :integer), **options)
```
Comment (kamipo) [inline activerecord/lib/active_record/migration/compatibility.rb]: We can simply call `super` if `column_options.reverse_merge!(type: :integer)` before if-else block.

Comment (kuzukuzu) [reply activerecord/lib/active_record/migration/compatibility.rb]: Done it. Thanks.

**Pull request merged**